thoth/adviser/sieves/index_enabled.py

```diff
@@ -50,14 +54,18 @@ def should_include(
 
     def run(self, package_version: PackageVersion) -> None:
         """Filter out package versions based on disabled Python package index."""
-        try:
-            is_enabled = self.graph.is_python_package_index_enabled(
-                package_version.index.url
-            )
-        except NotFoundError:
-            raise NotAcceptable(
-                f"Removing Python package version {package_version.to_tuple()!r} as used index is not registered"
-            )
+        if package_version.index.url in self._cached_records:
+            is_enabled = self._cached_records[package_version.index.url]
+        else:
+            try:
+                is_enabled = self.graph.is_python_package_index_enabled(
+                    package_version.index.url
+                )
+            except NotFoundError:
+                raise NotAcceptable(
```

Comment:
can we keep this information for later use?

Reply:
Done.